			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-009

Bug-Reported-by:	Corey Hickey <bugfood-ml@fatooh.org>
Bug-Reference-ID:	<134330ef-0ead-d73e-68eb-d58fc51efdba@fatooh.org>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/help-bash/2022-10/msg00025.html

Bug-Description:

Bash arithmetic expansion should allow `@' and `*' to be used as associative
array keys in expressions.

--- a/expr.c
+++ b/expr.c
@@ -1168,6 +1168,8 @@ expr_streval (tok, e, lvalue)
   /* [[[[[ */
 #if defined (ARRAY_VARS)
   aflag = tflag;	/* use a different variable for now */
+  if (shell_compatibility_level > 51)
+    aflag |= AV_ATSTARKEYS;
   v = (e == ']') ? array_variable_part (tok, tflag, (char **)0, (int *)0) : find_variable (tok);
 #else
   v = find_variable (tok);
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 8
+#define PATCHLEVEL 9
 
 #endif /* _PATCHLEVEL_H_ */
